src/test/java/io/vertx/test/core/HttpTest.java

```diff
@@ -1262,6 +1262,33 @@ private void testRequestHeaders(boolean individually) {
   }
 
   @Test
+  public void testRequestHeadersWithCharSequence() {
+    HashMap<CharSequence, String> headers = new HashMap<>();
+    headers.put(HttpHeaders.TEXT_HTML, "text/html");
+    headers.put(HttpHeaders.USER_AGENT, "User-Agent");
+    headers.put(HttpHeaders.APPLICATION_X_WWW_FORM_URLENCODED, "application/x-www-form-urlencoded");
+
+    server.requestHandler(req -> {
+      assertEquals(headers.size() + 1, req.headers().size());
+
+      headers.forEach((k, v) -> assertEquals(v, req.headers().get(k)));
```

Comment:
Is this really testing the new API? Or is it just calling the old methods that take a String?